R/db.R

```diff
@@ -297,6 +297,14 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns) {
   }
 
   write_column_hashed_data <- function(df, tblname, append = TRUE) {
+    ## Don't cache anything that is in the blacklist.
+    if (!is.null(blacklist)) {
+      if (any(is.na(blacklist))) {
+        df <- df[apply(df, 1, function(x) all(!(is.na(x)))),]
+      }
+      df <- df[apply(df, 1, function(x) all(!(x %in% blacklist))),]
```

Comment:
Again, try to avoid using `apply(df, 1, ...)` on dataframes because type coercion is very slow. Faster and cleaner is

```r
> blacklist <- list(1, 2)
> df <- data.frame(a = c(1,2,NA), b = 1:3, c = c(NA, 5, 7))
> df <- df[apply(Reduce(`|`, lapply(blacklist, `==`, df)), 1, Negate(any), na.rm = TRUE), ]
> df
   a b c
3 NA 3 7
```